koku/api/query_handler.py

```diff
@@ -298,3 +298,44 @@ def _get_filter(self, delta=False):
         filters.add(query_filter=end_filter)
 
         return filters
+
+    def filter_to_order_by(self, parameters):
+        """Remove group_by[NAME]=* and replace it with group_by[NAME]=X.
+
+        The parameters object contains a list of filters and a list of group_bys.
+
+        For example, if the parameters object contained the following:
+        group_by[X] = Y
+        group_by[Z] = *     # removes this line
+        filter[Z] = L
+        filter[X] = Y
+
+        The returned parameters object would contain lists that look like this:
+
+        group_by[X] = Y
+        group_by[Z] = L     # adds this line
+        filter[Z] = L
+        filter[X] = Y
+
+        Thereby removing the star when there is a filter provided.
+
+        Args:
+            parameters (QueryParameters): The parameters object
+
+        Returns:
+            parameters (QueryParameters): The parameters object
+
+        """
+        for group_by_key in parameters.parameters['group_by']:
```

Comment:
You should use a  `get` on the object otherwise you could get a *KeyError* when accessing a key that is not in the query parameters.
`for group_by_key in parameters.parameters.get('group_by', {}):`

Reply:
If group_by_key is not in the parameters wouldn’t we want an error to be thrown?
https://github.com/project-koku/koku/blob/master/koku/api/query_params.py#

Reply:
Any query can be run without a group_by present and defense coding practices have you not depend on another portion of code which could change outside of this to initialize group_by in the parameters object.

Reply:
I removed this line

Reply:
@chambridge the line should be added now, is it still not visible?

Reply:
Yes, I see the update. I made a comment here: https://github.com/project-koku/koku/pull/1552/files#r368004234 that can be addressed, but I'll resolve this comment.